### ISSUE TITLE
Multifidelity dataset and data feeding/sampling

### ISSUE THREAD

**Issue by edgmin** [lululxvi/deepxde#450]
Hello Lu,

First of all many thanks for your efforts and for making deepxde available, it is really useful.

I am trying to reimplement the multifidelity network alone in Pytorch. As I am not very familiar with TensorFlow, I am struggling a bit to properly understand. I would be happy if you could give me some advice.

As I understood, at least when using tensorflow backend, the number of epochs corresponds to a single pass over all samples on a dataset. In the case of multifidelity networks, we have the low and multifidelity datasets (forget about the experimental dataset for now) which, if I understood well, are employed simultaneously to train the low and high fidelity 'portions' of the multifidelity model. My quesiton is, what do you do when for example one dataset has less points than the other?

Takes this as an example: the high fidelity network need as input the 'low fidelity prediction' of a low fidelity sample 'concatenated' with a high fidelity sample. What if one of the datasets, lets say the low fid has already used all samples? Do you starting sampling from the dataset again till the total number of iterations (epochs over the dataset) is done? Do you use sample replacement or do you use all samples them start sampling from the dataset again?

Thanks in advance.

**Comment by lululxvi** [comment]
The total loss is the sum of two losses: the low-fidelity loss and the high-fidelity loss. So, these two datasets are independent, and their sizes don't matter.

If you would like to use PyTorch, you only need to translate the Tensorflow code (https://github.com/lululxvi/deepxde/blob/master/deepxde/nn/tensorflow_compat_v1/mfnn.py) to Pytorch. Nothing else will be changed. You can also look at the FNN code of Tensoflow and Pytorch for a comparison.
